### PR TITLE
[5.2] Model accessors used in custom relationship columns

### PR DESCRIPTION
**Pull request by kirkbushell**
There is a bug that exists whereby if a model's accessor returns a value other than primitive (such as an array, due to JSON decoding.etc.), and that same column is used for a join, then the relationship completely fails as an array cannot be used as part of the relationship value.

This PR solves that by ensuring only the raw model attribute is used, not the malformed value due to an accessor.

### CODE REVIEW

**Comment by taylorotwell** [comment]
This looks better. But I think you would still have errors on non-eager loaded relations. For example, if I have a simple `HasMany` and the key value is mutated I think that would still error?

**Comment by kirkbushell** [comment]
@taylorotwell that makes sense. I'll sort that out shortly.

**Comment by kirkbushell** [comment]
@taylorotwell sorted.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -2620,6 +2630,19 @@ public function getAttribute($key)
     }
 
     /**
+     * Retrieve a value (raw) for an attribute from this model.
+     *
+     * @param string $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: cs ;)

Comment (kirkbushell) [inline src/Illuminate/Database/Eloquent/Model.php]: ?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: we need two spaces please, each time

Comment (kirkbushell) [inline src/Illuminate/Database/Eloquent/Model.php]: Two spaces where? It passes the style ci tests...

Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: I know, I've not written that fixer yet. ;)

Comment (kirkbushell) [inline src/Illuminate/Database/Eloquent/Model.php]: Okay no problem - so your request still isn't very clear. Where do you want 2 spaces?

Comment (taylorotwell) [inline src/Illuminate/Database/Eloquent/Model.php]: Don't worry about it.

**Comment by kirkbushell** [comment]
@taylorotwell good to go then?

**Comment by taylorotwell** [comment]
This is still broken when just using normal relationships. For example if I do something like this it's broken: 

```
class UserId {}

class Flight extends Eloquent {
    public function getUserIdAttribute()
    {
        return new UserId($this->attributes['user_id']);
    }
    public function user()
    {
        return $this->belongsTo(App\User::class, 'user_id');
    }
}

Route::get('/', function () {
    $flight = Flight::find(1);
    $flight->user;
});
```

**Comment by taylorotwell** [comment]
Closing pending further activity.

**Comment by kirkbushell** [comment]
@taylorotwell sorry, will get to this in time. Hopefully this weekend :)

**Pull request closed without merging**